exchanges/bybit/bybit_wrapper.go

```diff
@@ -0,0 +1,1844 @@
+				TransferID:      strconv.FormatInt(w[i].ID, 10),
+				Currency:        w[i].Coin,
+				Amount:          w[i].Amount,
+				Fee:             w[i].Fee,
+				CryptoToAddress: w[i].Address,
+				CryptoTxID:      w[i].TxID,
+				Timestamp:       w[i].UpdatedAt,
+			}
+		}
+		return withdrawHistory, nil
+	default:
+		return nil, fmt.Errorf("%s %w", a, asset.ErrNotSupported)
+	}
+}
+
+// GetRecentTrades returns the most recent trades for a currency and asset
+func (by *Bybit) GetRecentTrades(ctx context.Context, p currency.Pair, assetType asset.Item) ([]trade.Data, error) {
+	var resp []trade.Data
+
+	formattedPair, err := by.FormatExchangeCurrency(p, assetType)
+	if err != nil {
+		return nil, err
+	}
+
+	switch assetType {
+	case asset.Spot:
+		tradeData, err := by.GetTrades(ctx, formattedPair.String(), 0)
+		if err != nil {
+			return nil, err
+		}
+
+		for i := range tradeData {
+			resp = append(resp, trade.Data{
+				Exchange:     by.Name,
+				CurrencyPair: p,
+				AssetType:    assetType,
+				Price:        tradeData[i].Price,
+				Amount:       tradeData[i].Volume,
+				Timestamp:    tradeData[i].Time,
+			})
+		}
+
+	case asset.CoinMarginedFutures, asset.Futures:
+		tradeData, err := by.GetPublicTrades(ctx, formattedPair, 0)
+		if err != nil {
+			return nil, err
+		}
+
+		for i := range tradeData {
+			resp = append(resp, trade.Data{
+				Exchange:     by.Name,
+				CurrencyPair: p,
+				AssetType:    assetType,
+				Price:        tradeData[i].Price,
+				Amount:       tradeData[i].Qty,
+				Timestamp:    tradeData[i].Time,
+			})
+		}
+
+	case asset.USDTMarginedFutures:
+		tradeData, err := by.GetUSDTPublicTrades(ctx, formattedPair, 0)
+		if err != nil {
+			return nil, err
+		}
+
+		for i := range tradeData {
+			resp = append(resp, trade.Data{
+				Exchange:     by.Name,
+				CurrencyPair: p,
+				AssetType:    assetType,
+				Price:        tradeData[i].Price,
+				Amount:       tradeData[i].Qty,
+				Timestamp:    tradeData[i].Time,
+			})
+		}
+
+	case asset.USDCMarginedFutures:
+		tradeData, err := by.GetUSDCLatestTrades(ctx, formattedPair, "PERPETUAL", 0)
+		if err != nil {
+			return nil, err
+		}
+
+		for i := range tradeData {
+			resp = append(resp, trade.Data{
+				Exchange:     by.Name,
+				CurrencyPair: p,
+				AssetType:    assetType,
+				Price:        tradeData[i].OrderPrice,
+				Amount:       tradeData[i].OrderQty,
+				Timestamp:    tradeData[i].Timestamp.Time(),
+			})
+		}
+
+	default:
+		return nil, fmt.Errorf("%s %w", assetType, asset.ErrNotSupported)
+	}
+
+	if by.IsSaveTradeDataEnabled() {
+		err := trade.AddTradesToBuffer(by.Name, resp...)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	sort.Sort(trade.ByDate(resp))
+	return resp, nil
+}
+
+// GetHistoricTrades returns historic trade data within the timeframe provided
+func (by *Bybit) GetHistoricTrades(ctx context.Context, p currency.Pair, assetType asset.Item, timestampStart, timestampEnd time.Time) ([]trade.Data, error) {
```

Comment:
Since this exchange only supports recent trades based on their docs without being able to specify time ranges, can set this to `return nil, common.ErrNotSupported`